### PR TITLE
[Share 2.0] Add missing post_update_permissions hook

### PR DESCRIPTION
**Pull request by rullzer**
@nickvergessen as prommised.

CC: @schiesbn @nickvergessen @PVince81

### CODE REVIEW

**Comment by PVince81** [comment]
Code looks good :+1: 

I'll have a look at the failing JS tests separately, seems to be random.

File: lib/private/share20/manager.php [hunk 1]
```diff
@@ -616,6 +616,18 @@ public function updateShare(\OCP\Share\IShare $share) {
...
 
+		if ($share->getPermissions() !== $originalShare->getPermissions()) {
+			\OC_Hook::emit('OCP\Share', 'post_update_permissions', array(
+				'itemType' => $share->getNode() instanceof \OCP\Files\File ? 'file' : 'folder',
+				'itemSource' => $share->getNode()->getId(),
+				'shareType' => $share->getShareType(),
+				'shareWith' => $share->getSharedWith(),
+				'uidOwner' => $share->getSharedBy(),
+				'permissions' => $share->getPermissions(),
+				'path' => $share->getNode()->getPath(),
```
Comment (nickvergessen) [inline lib/private/share20/manager.php]: While I prefer this one, the old hook sent the local path, not the absolute path

### 8.2
```
Info 	sharing_log 	User admin shared "/test2" with the user test1, permissions: READ UPDATE CREATE DELETE SHARE [CLIENT_IP: 127.0.0.1] 	2016-02-04T12:11:01+00:00 
Info 	sharing_log 	User admin updated the permissions for "/test/test2" for the user test1, permissions: READ UPDATE CREATE DELETE [CLIENT_IP: 127.0.0.1] 	2016-02-04T12:11:02+00:00
```

### This PR
```
Info 	admin_audit 	User admin shared "/test2" with the user test1, permissions: READ UPDATE CREATE DELETE SHARE [CLIENT_IP: 127.0.0.1] 	2016-02-04T12:11:53+00:00 
Info 	admin_audit 	User admin updated the permissions for "/admin/files/test/test2" for the user test1, permissions: READ UPDATE CREATE DELETE [CLIENT_IP: 127.0.0.1] 	2016-02-04T12:11:54+00:00
```

Comment (rullzer) [inline lib/private/share20/manager.php]: fixed... should be the relative path now...

**Comment by PVince81** [comment]
Triple unlucky:
- JSUnit test unrelated: https://github.com/owncloud/core/pull/22130 (please review)
- Swift test: "MySQL has gone away"
- Smashbox: hudson.remoting.ChannelClosedException: channel is already closed

**Comment by nickvergessen** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
needs rebase

**Comment by rullzer** [comment]
Rebased

**Pull request merged**